From f351cd4e68aec726907f849e28a41f368d0569ea Mon Sep 17 00:00:00 2001
From: Arc Wang <arcwang@google.com>
Date: Thu, 11 Mar 2021 12:10:18 +0800
Subject: [PATCH 09/19] [Q_asb_2021-06] Hide non-system overlay window on
 ActivityPicker

To improve security.

Bug: 181962311
Test: manual
      Show an AlertDialog and observe if it will hide after below command.
      adb shell am start -a android.intent.action.PICK_ACTIVITY -n com.android.settings/.ActivityPicker
Change-Id: I6e2845cc19dc012cba2933318a067bbb8db90a23
(cherry picked from commit 3b4853e109a5d95656f73d31dace4099d5dfbca7)
(cherry picked from commit eac0ba87d0bfc700ab2e565ff673175f460147c6)
---
 src/com/android/settings/ActivityPicker.java | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/com/android/settings/ActivityPicker.java b/src/com/android/settings/ActivityPicker.java
index ae61944cbe..f75ce3740a 100644
--- a/src/com/android/settings/ActivityPicker.java
+++ b/src/com/android/settings/ActivityPicker.java
@@ -16,6 +16,8 @@
 
 package com.android.settings;
 
+import static android.view.WindowManager.LayoutParams.SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS;
+
 import android.app.Activity;
 import android.content.Context;
 import android.content.DialogInterface;
@@ -71,6 +73,8 @@ public class ActivityPicker extends AlertActivity implements
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
+
+        getWindow().addPrivateFlags(SYSTEM_FLAG_HIDE_NON_SYSTEM_OVERLAY_WINDOWS);
         
         final Intent intent = getIntent();
         
-- 
2.35.1

